data/notebooks/Lab/Excel_validation_logic.ipynb

```diff
@@ -0,0 +1,827 @@
+       "4             NaN     Moll       Unknown                  Malaysia     NaN   \n",
+       "5             NaN     Moll       Unknown  United States of America     NaN   \n",
+       "6             NaN     Moll       Unknown                     Japan     NaN   \n",
+       "7             NaN     Moll       Unknown                     India     NaN   \n",
+       "\n",
+       "2  latitude  longitude  2010_tons  ...  2015_tons  2016_tons  2017_tons  \\\n",
+       "3       NaN        NaN       2400  ...       2522       2547       2572   \n",
+       "4       NaN        NaN       1300  ...       1366       1380       1394   \n",
+       "5       NaN        NaN       1000  ...       1050       1061       1072   \n",
+       "6       NaN        NaN        730  ...        767        775        783   \n",
+       "7       NaN        NaN        490  ...        515        520        525   \n",
+       "\n",
+       "2  2018_tons  2019_tons  2020_tons  ...  data_source  comments  ...  \n",
+       "3       2598       2624       2650  NaN          NaN   aoeuijj  NaN  \n",
+       "4       1408       1422       1436  NaN          NaN       NaN  NaN  \n",
+       "5       1083       1094       1105  NaN          NaN       NaN  NaN  \n",
+       "6        791        799        807  NaN          NaN       NaN  NaN  \n",
+       "7        530        535        540  NaN          NaN       NaN  NaN  \n",
+       "\n",
+       "[5 rows x 24 columns]"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "data_validation.validate(df).head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "867a8fdd",
+   "metadata": {},
+   "source": [
+    "## Locations validation. \n",
+    "\n",
+    "Check if location data is in the correct format, according to the following logic:  \n",
+    "(note: Country info and coordinates are validated in the previous step)  \n",
+    "\n",
+    "- if location_type is Unknown, it should not contain info on address and coordinates (only country)\n",
+    "- if location_type is Country of production or Origin country, also no info on address and coordinates\n",
+    "- if location type is Point of production, Aggregation point or Origin suplier facility, it MUST contain address or coordinates\n",
+    "\n",
+    "The result is a log of the outcomes (correct or type of error) for each entry."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 97,
+   "id": "14e73d61",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def location_validation(df):\n",
+    "    for l in range(len(df)):\n",
+    "        if 'country' in df.iloc[l]['location_type'].lower():\n",
+    "            if not pd.isna(df.iloc[l]['address']) or not pd.isna(df.iloc[l]['latitude']) or not pd.isna(df.iloc[l]['longitude']):\n",
+    "                print(f'LOCATION ERROR ON ENTRY {l+1}: address/latitude/longitude not allowed')\n",
```

Comment:
I wouldn't say we should provide this as an error but as a warning - the user can provide that information but it's not required/ used for the geocoding or maybe there is a case were the user can provide this information and maybe we can suggest the option of changing the location type?